### PR TITLE
Feat: Add base for rendering workspaces items in the settings sidebar

### PR DESCRIPTION
**Pull request by Mikehrn**
Abstracted this from the adding general workspaces branches, so it doesn't blocks adding other pages.
Fetches all the workspaces and adds them to the settings sidebar. Doesn't add any links to specific pages but they will be added in the related PRs. Will only be shown when the user has actual workspace so it won't show for any existing users (as they wont have workspaces)

### CODE REVIEW

**Review by andrewwallacespeckle — DISMISSED** [review]
LGTM - Great to get the first PR out for workspaces. Because it's the first and we will base a lot on this structure, I'd wait for @fabis94 to review before we merge this one.

File: packages/frontend-2/components/settings/Dialog.vue
Comment (fabis94) [inline packages/frontend-2/components/settings/Dialog.vue:51]: https://www.notion.so/speckle/General-Web-Code-Review-Guide-1f2bf3dfea634460b3778ac3cf4aa1bb?pvs=4#af99b815b2a6430da927ca10a1eb9007

File: packages/frontend-2/components/settings/Dialog.vue
Comment (fabis94) [inline packages/frontend-2/components/settings/Dialog.vue:112]: You need to introduce and start using a similar composable to `useIsAutomateModuleEnabled()` that check the workspaces feature flag.

Cause this query should not even be firing with that FF off - otherwise the server is gonna throw an error. So the workspace settings UI & also all the relevant GQL operations should be blocked by the feature flag.

**Review by fabis94 — CHANGES_REQUESTED** [review]

File: packages/frontend-2/components/settings/Dialog.vue
Comment (Mikehrn) [reply packages/frontend-2/components/settings/Dialog.vue]: Good point; didn't know they would throw an error when not available. I've adjusted this to use a feature flag, following the example of automate

**Review by fabis94 — APPROVED** [review]

**Pull request merged**
